product_supplierinfo_barcode/models/supplierinfo.py

```diff
@@ -0,0 +1,18 @@
+from odoo import fields, models
+
+
+class SupplierInfo(models.Model):
+    _inherit = "product.supplierinfo"
+
+    barcode = fields.Char(
+        copy=False,
+        help="Article number used by supplier for product identification.",
+    )
+
+    _sql_constraints = [
+        (
+            "barcode_uniq",
+            "unique(barcode)",
```

Comment:
I think this constraint is too restrictive, as several suppliers may have the same barcode. It should be a combination of unique company, seller and barcode